### ISSUE TITLE
Override Java Memory setting resets to a extremely low ram everytime crashing game due to not enough ram (happens in the latest master build "switch to preact")

### ISSUE THREAD

**Issue by Jordan231111** [killpowa/Twickt-Launcher#1276]
**Describe the bug**
Everytime opening GDLauncher, the override java memory setting auto sets to something like 2.5000948141217806 MB instead of what i have it at like 4GB or 6GB

**To Reproduce**
1. Go turn on overide java arguments in ur modpack
2. start the modpack. do whatever, close game
3. NExt time open gd launcher, game crashes because override java memory automatically set to 2.5 MB ram
![image](https://user-images.githubusercontent.com/79342877/153080826-dc8fcd89-442b-4fe0-8c6d-ab0e77efa18c.png)

**Expected behavior**
For my ram setting to stay at 4 or 6 GIGABYTE
*
Windows 10 java 17 and java 18
basically any modpack or java version and windows 10
**Additional context**

**Comment by AshtakaOOf** [comment]
can you give more information? like logs.

**Comment by TrickDevelopment** [comment]
I am having this same issue.
It occurs when you try to override the amount of memory java will use.
If you change the amount of memory it works fine.
But as soon as you leave the screen where you change the memory, the memory becomes a really small number like 3.6000290...

This video demonstrates the issue
https://user-images.githubusercontent.com/85457129/153669458-aeb28230-1722-4a26-87a3-134de68238de.mp4

**Comment by Jordan231111** [comment]
yes there is also no log cuz it just happens when opening gd launcher

**Comment by AshtakaOOf** [comment]
Can you try the latest version ?

**Comment by TrickDevelopment** [comment]
The thing I did to fix this, was to switch from the beta version to the stable version.

**Comment by Jordan231111** [comment]
yes but stable version is way out of date and does not work as stable version is broken due to api changes. the beta at least somewhat works with downloading mods

**Comment by AshtakaOOf** [comment]
yes Davide switched to the new api it seems to work pretty well for now at least.
Also wait a little Davide is working on the Ram slider but the slider work for now,
you can actually change the ram amount and it work but when you open the manage panel of an instance it will look like that but it actually took the desired ram amount so it will launch with the right amount of ram so for now use gdl like that and wait

**Comment by AdamEternal** [comment]
Same issue. There's a little more that can happen, with some fiddling.

Override Java Memory: off -> open instance manager -> Override Java Memory: on -> The override setting starts on the default memory, following expected behaviour. If you enable and disable the setting at this point, the setting remains unchanged.

If, after doing that and closing, you re-open it, it'll be set to _Infinity MB._ (It literally says that! I didn't know that was a thing!) Opening and closing the instance manager at this point does not change this setting.

After changing the memory override with the bar, closing and opening the instance manager will result in the usual bug: super small MB setting.

Notably, the amount of the tiny MB depends on where you set your bar before closing the manager. I have 16 GB (16.384 GB exactly). If I set the bar to the max (which is _not_ infinity?), I'll get 4 MB after re-opening the manager.

These are the amounts I get, at the power-of-two increments:

- 16,384 MB -> 4 MB
- 8,192 MB -> 3 MB
- 3,096 MB -> 2 MB
- 2,048 MB -> 1 MB
- 1,024 MB -> The setting turns itself off at 1,024 MB.

Here are some other, arbitrary amounts and their results, to the eighth decimal place:

- 5,793 MB -> 2.50009494... MB
- 3,822 MB -> 1.90011206... MB
- 1,097 MB -> 0.09934781... MB

Quick edit: I'm using the Beta build.

**Comment by MokahTGS** [comment]
On Beta 1.1.22 Beta.1 and having the same issue.  This makes playing any larger modpack impossible.

**Comment by KajesLorian** [comment]
Having the same issue on 1.1.22 Beta.1
Trying to make a modpack for a new server and am worried my players won't want to have to manually set there memory every single time they want to play.

**Comment by MokahTGS** [comment]
This is a huge problem as no major modpack can be used with this launcher

**Comment by TrickDevelopment** [comment]
The best advice I could give to anyone having this problem is to switch back to the stable version. The beta version can contain bugs as its a beta version. So if you have this problem, try switching back to the stable version of the launcher.

**Comment by MokahTGS** [comment]
> The best advice I could give to anyone having this problem is to switch back to the stable version. The beta version can contain bugs as its a beta version. So if you have this problem, try switching back to the stable version of the launcher.

The stable version is so out of date it does not work with 1.18+  There needs to be an update ASAP.

**Comment by Jordan231111** [comment]
^^ so only possible method is to change default ram in launcher settings or manually change it each time in override java memory. the stable version is unusable period

**Comment by AshtakaOOf** [comment]
> > The best advice I could give to anyone having this problem is to switch back to the stable version. The beta version can contain bugs as its a beta version. So if you have this problem, try switching back to the stable version of the launcher.
> 
> The stable version is so out of date it does not work with 1.18+ There needs to be an update ASAP.

You see currently there is no active devs this is an open source project and the main dev only get money from donation and it isn't enough to work full time on this project

**Comment by Jordan231111** [comment]
I miss the days when gdlauncher worked pretty much perfectly. sad that curseforge had to change their api. totally destroyed this launcher

**Comment by FreeFull** [comment]
Encountering this bug in v1.1.23

**Comment by embeddedt** [comment]
I had to upgrade to 1.1.23 because of the CurseForge API breakage and I'm now experiencing this as well. Unfortunately it makes the launcher largely unusable.

@killpowa What are the steps to get a dev environment going? I tried `npm install`, `npm run dev`, and the launcher did open, but it seemed that none of the changes I made in the code were being reflected even if I closed everything and re-ran `npm run dev`.

**Comment by embeddedt** [comment]
Bump, as far as I can tell, this remains an issue in 1.1.24.

**Comment by Diztend** [comment]
I changed the ram in global settings and nothing launches anymore, even after I revert

**Comment by spaghetus** [comment]
I just had the latest Flatpak version of GDLauncher on Linux try to start Minecraft with `-XmxInfinity`, I'm not sure if that's related

**Comment by WolfBV** [comment]
@spaghetus Assuming you haven’t already solved your issue, I experienced the same -Xmxinfinity and fixed it by deleting a config file with gdlauncher’s jvm arguments somewhere in gdlauncher_next>IndexedDB. I also experienced -XmxInfinitym and unrecognized vm argument “, which were fixed by doing this.
